src/Equinox.MessageDb/MessageDb.fs

```diff
@@ -0,0 +1,415 @@
+[<NoComparison; NoEquality; RequireQualifiedAccess>]
+type CachingStrategy =
+    /// Retain a single 'state per streamName.
+    /// Each cache hit for a stream renews the retention period for the defined <c>window</c>.
+    /// Upon expiration of the defined <c>window</c> from the point at which the cache was entry was last used, a full reload is triggered.
+    /// Unless <c>LoadOption.AllowStale</c> is used, each cache hit still incurs a roundtrip to load any subsequently-added events.
+    | SlidingWindow of ICache * window : TimeSpan
+    /// Retain a single 'state per streamName
+    /// Upon expiration of the defined <c>period</c>, a full reload is triggered.
+    /// Unless <c>LoadOption.AllowStale</c> is used, each cache hit still incurs a roundtrip to load any subsequently-added events.
+    | FixedTimeSpan of ICache * period : TimeSpan
+    /// Prefix is used to segregate multiple folds per stream when they are stored in the cache.
+    /// Semantics are identical to <c>SlidingWindow</c>.
+    | SlidingWindowPrefixed of ICache * window : TimeSpan * prefix : string
+
+type MessageDbCategory<'event, 'state, 'context>(resolveInner, empty) =
+    inherit Equinox.Category<'event, 'state, 'context>(resolveInner, empty)
+    new (   context : MessageDbContext, codec : IEventCodec<_, _, 'context>, fold, initial,
+            [<O; D(null)>]?caching,
+            [<O; D(null)>]?access) =
+        let inner = Category<'event, 'state, 'context>(context, codec, ?access = access)
+        let readCacheOption =
+            match caching with
+            | None -> None
+            | Some (CachingStrategy.SlidingWindow (cache, _))
+            | Some (CachingStrategy.FixedTimeSpan (cache, _)) -> Some (cache, null)
+            | Some (CachingStrategy.SlidingWindowPrefixed (cache, _, prefix)) -> Some (cache, prefix)
+        let folder = Folder<'event, 'state, 'context>(inner, fold, initial, ?readCache = readCacheOption)
+        let category : ICategory<_, _, 'context> =
+            match caching with
+            | None -> folder :> _
+            | Some (CachingStrategy.SlidingWindow (cache, window)) ->
+                Caching.applyCacheUpdatesWithSlidingExpiration cache null window folder Token.supersedes
+            | Some (CachingStrategy.FixedTimeSpan (cache, period)) ->
+                Caching.applyCacheUpdatesWithFixedTimeSpan cache null period folder Token.supersedes
+            | Some (CachingStrategy.SlidingWindowPrefixed (cache, window, prefix)) ->
+                Caching.applyCacheUpdatesWithSlidingExpiration cache prefix window folder Token.supersedes
+        let resolveInner streamIds = struct (category, StreamName.Internal.ofCategoryAndStreamId streamIds, ValueNone)
+        let empty = struct (context.TokenEmpty, initial)
+        MessageDbCategory(resolveInner, empty)
+
+
+type MessageDbConnector(
+    connectionString : string, [<O; D(null)>]?readConnectionString : string,
```

Comment:
add comment on `readConnectionString` 
```
// Can be used to divert reads to a replica
// Conflicts detected on write trigger a resync, reading via the `connectionString` to maximize the freshness of the data for the retry
```
this also makes me think a `LoadOption.RequireLeader`  would be a good feature. It would mean:
- force reading from leader on ES (because you have reason to believe a follower read has a high chance of being out of date)
- force that flag even for a normal load in MDB
- force the 'quorum read' param when talking to DDB (only plumbed through in a very recent version of FSharp.Aws.DDB) (this costs latency and more RUs)

Reply:
Great idea. Do you know if there's a way to make that pop up in auto completes?

Reply:
Clarification: It think both adding the comment, and the LoadOption are great ideas. I'm asking whether you know the magic syntax annotation to make a comment like that pop up in auto complete when I start typing "readC"

Reply:
Sadly there is no easy way. I used to put `///` over the arg value and was mysified why it did not work - sadly that's not implemented yet in fsc at param name level). Intellisense requires a full csc style xmldoc comment with parameter level comments. But in general I personally use Rider and am constantly hopping in to see the `//` comments via `gd` (the nugets have sourcelink)

Reply:
Logged https://github.com/jet/equinox/issues/340